examples/gum/hook_open/src/lib.rs

```diff
@@ -1,30 +1,41 @@
-#![feature(c_variadic)]
-use frida_gum::{
-    interceptor::{Interceptor},
-    Gum,
-    Module, NativePointer
-};
-use lazy_static::lazy_static;
 use ctor::ctor;
-use libc::{c_void, c_int, c_char};
+use frida_gum::{interceptor::Interceptor, Gum, Module, NativePointer};
+use lazy_static::lazy_static;
+use libc::{c_char, c_int, c_void};
+use null_fn::null_fn;
 
 lazy_static! {
     static ref GUM: Gum = unsafe { Gum::obtain() };
 }
 
-unsafe extern "C" fn open_detour(
-    name: *const c_char, 
-    flags: c_int,
-) -> c_int {
-    println!("open_detour: {}", std::ffi::CStr::from_ptr(name).to_str().unwrap());
-    let res = libc::open(name, flags);
-    res
+type OpenFunc = unsafe extern "C" fn(*const c_char, flags: c_int) -> c_int;
+
+/// Should use SyncUnsafeCell in most scenarios to avoid safeness issues (static mut)
+#[null_fn]
+static mut ORIGINAL_OPEN: OpenFunc = null();
```

Comment:
Why are we using `null_fn`? What is wrong with an `Mutex<Option<`?

Reply:
imo having a mutex doesn't make sense as it's only changed before the function is active, same for option, but will use it and add a comment.